src/psbt.cpp

```diff
@@ -207,7 +209,8 @@ size_t CountPSBTUnsignedInputs(const PartiallySignedTransaction& psbt) {
 
 void UpdatePSBTOutput(const SigningProvider& provider, PartiallySignedTransaction& psbt, int index)
 {
-    const CTxOut& out = psbt.tx->vout.at(index);
+    CMutableTransaction& tx = *Assert(psbt.tx);
```

Comment:
nit: to be explicit
```suggestion
    CMutableTransaction& tx = *Assert(psbt.tx.has_value());
```
as were [suggested](https://github.com/bitcoin/bitcoin/pull/20671#issuecomment-748046093) by @jnewbery.

Reply:
That wouldn't compile. Also, what is the benefit of it?

Reply:
> That wouldn't compile.

Indeed :man_facepalming: 

Sorry for noice.